mobile/src/main/scala/com/github/shadowsocks/MainActivity.scala

```diff
@@ -400,7 +417,6 @@ class MainActivity extends Activity with ServiceBoundContext with Drawer.OnDrawe
     val currentFragment = getFragmentManager.findFragmentById(R.id.fragment_holder).asInstanceOf[ToolbarFragment]
     if (!currentFragment.onBackPressed())
       if (currentFragment.isInstanceOf[ProfilesFragment]) super.onBackPressed() else {
-        displayFragment(new ProfilesFragment)
```

Comment:
Seems like this is the only change that's relevant to your title. I will merge this change.